assignments/arrays.js

```diff
@@ -63,34 +63,72 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
+function findCar(arr, index) {
+  console.log(`Car ${arr[index].id} is a ${arr[index].car_year}  ${arr[index].car_make} ${arr[index].car_model}`)
+}
 
+console.log(findCar(inventory, 32));
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory[inventory.length-1];
+console.log(`${lastCar.car_make} ${lastCar.car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+function compare(i, j) {
+    if (i.car_model < j.car_model)
+      return -1;
+    if (i.car_model > j.car_model)
+      return 1;
+    return 0;
+  }
+
+console.log(inventory.sort(compare));
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+function findCarYear(arr) {
+  let carYears = [];
+  for (i=0; i<arr.length; i++) {
+    carYears.unshift(arr[i].car_year);
+  }
+  return carYears;
+}
+
+console.log(findCarYear(inventory));
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+function oldCarArr(arr)  {
+  let oldCars = [];
+  for (i=0; i<arr.length; i++) {
+    if (arr[i].car_year < 2000) {
+      oldCars.unshift(arr[i].car_year)
+    }
+  }
+  return oldCars;
```

Comment:
Instead of returning an array of old cars, just return the length!

```suggestion
  return oldCars.length;
```